v2/spiffetls/dial.go

```diff
@@ -25,10 +37,73 @@ func (c *Conn) Close() error {
 //
 // This is the same as DialWithMode using the MTLSClient mode.
 func Dial(ctx context.Context, network, addr string, authorizer tlsconfig.Authorizer, options ...DialOption) (*Conn, error) {
-	panic("not implemented")
+	return DialWithMode(ctx, network, addr, MTLSClient(authorizer), options...)
 }
 
 // DialWithMode creates a TLS connection using the specified mode.
 func DialWithMode(ctx context.Context, network, addr string, mode DialMode, options ...DialOption) (*Conn, error) {
-	panic("not implemented")
+	m := mode.get()
+	var err error
+	source := *m.source
```

Comment:
This should be a `*workloadapi.X509Source` right?
```suggestion
	source := m.source
```